netrd/distance/jaccard_distance.py

```diff
@@ -0,0 +1,46 @@
+"""
+jaccard_distance.py
+--------------
+
+Graph distance based on https://scikit-learn.org/stable/modules/generated/sklearn.metrics.jaccard_similarity_score.html
+
+author: David Saffo
+email: saffo.d@husky.neu.edu
+Submitted as part of the 2019 NetSI Collabathon.
+
+"""
+
+from .base import BaseDistance
+from sklearn.metrics import jaccard_similarity_score
+import networkx as nx
+
+
+
+class JaccardDistance(BaseDistance):
+    def dist(self, G1, G2):
+        """Computes the average jaccard index between two sparse matrices
+        
+        implementation details here:
+        https://scikit-learn.org/stable/modules/model_evaluation.html#jaccard-similarity-score
+        and here:
+        https://scikit-learn.org/stable/modules/generated/sklearn.metrics.jaccard_similarity_score.html
+
+        Params
+        ------
+
+        G1, G2 (nx.Graph): two networkx graphs to be compared.
+
+        Returns
+        -------
+
+        dist (float): the distance between G1 and G2.
+
+        """
+        adj1 = nx.to_numpy_matrix(G1)
+        adj2 = nx.to_numpy_matrix(G2)
+        dist = jaccard_similarity_score(adj1,adj2)
```

Comment:
I like that we have something as interpretable as Jaccard. However, this function seems to return a similarity score, as opposed to a distance score. There are two changes we need to make before we can merge this:

1. Return a distance score, as opposed to a similarity score. If the output of this function is between 0 and 1, then we can just return `1 - jaccard_similarity_score(adj1, adj2)`. For example, if someone called `JaccardDistance.dist(G, G)`, over two copies of the same graph `G`, the output should be zero.

2. We need to add scikit-learn to the `netrd/requirements.txt` file. Please add the line `scikit-learn==0.18.2`. This will allow the tests to run (right now they are failing). If we need a different version, then that can be changed later.

Reply:
Shouldn't it be `scikit-learn >= 0.18.2`, not `scikit-learn == 0.18.2`?

Reply:
@sdmccabe I think so. Also @dsaffo Travis is failing b/c `numpy` hasn't been imported.